server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationMetadata.java

```diff
@@ -337,12 +336,12 @@ public VotingConfiguration(Set<String> nodeIds) {
         }
 
         public VotingConfiguration(StreamInput in) throws IOException {
-            nodeIds = Collections.unmodifiableSet(Sets.newHashSet(in.readStringArray()));
+            nodeIds = Set.copyOf(in.readStringList());
```

Comment:
NIT:
```suggestion
            nodeIds = Set.of(in.readStringArray());
```
I believe this way we would not need to create intermediate list. Array would still be copied.

Reply:
Ah right yes even nicer.